src/v2i-hub/TimPlugin/src/TimPlugin.cpp

```diff
@@ -0,0 +1,438 @@
+#include "TimPlugin.h"
+#include <WGS84Point.h>
+#include "TimeHelper.h"
+#include "XmlCurveParser.h"
+#include "VehicleLocate.h"
+#include <tmx/messages/IvpDmsControlMsg.h>
+#include <tmx/j2735_messages/TravelerInformationMessage.hpp>
+
+using namespace std;
+using namespace tmx::messages;
+using namespace tmx::utils;
+
+namespace TimPlugin {
+
+
+
+
+/**
+ * Construct a new EpcwPlugin with the given name.
+ *
+ * @param name The name to give the plugin for identification purposes
+ */
+TimPlugin::TimPlugin(string name) :
+		PluginClient(name) {
+
+	AddMessageFilter < BsmMessage > (this, &TimPlugin::HandleBasicSafetyMessage);
```

Comment:
Does TIM plugin needs to handle BSMs? Delete this if not needed.

Reply:
Addressed